### DISCUSSION TITLE
Successful build: S2-WL-ST + S5-EH1P6K-L - cost effective

### DISCUSSION THREAD

**Discussion by 0rangutan** [alienatedsec/solis-ha-modbus-cloud#24] [Show and tell]
@alienatedsec - Just a quick note to say thanks for providing such a great guide.
I have implemented this for my Solis inverter using the 12-way version of the consumer unit box that you suggested (tight but just fits): https://www.amazon.co.uk/dp/B09VKTGGXQ?ref=ppx_yo2ov_dt_b_product_details&th=1
plus this PoE switch: https://www.amazon.co.uk/dp/B0B7KYY2W1?ref=ppx_yo2ov_dt_b_product_details&th=1
Waveshare devices cheap at PiHut (£48 for two), switch £17, case £15, cables £4, existing Solis dongle, Exceedconn £9, **total £93**.
All running smoothly for a week now.
![20231116_202021](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/3738389/4c4f9de5-3a41-48be-b9b9-4accc014cf54)

**Comment by alienatedsec** [comment]
Looks neat @0rangutan thanks for sharing 

Would you mind creating a PR to update on pricing and items used?

**Reply by 0rangutan** [reply]
Sorry - no idea how to do that I'm afraid.
The PoE switch and Consumer Unit are linked above.  The latter includes the DIN rail so that can be dropped from the BOM.  It also came with basic compression glands for a possible further saving.
The better priced Waveshare RS485 PoE modules came from PiHut here: https://thepihut.com/products/rs485-to-rj45-ethernet-module

**Reply by alienatedsec** [reply]
@0rangutan thats ok

You could fork my repo, make updates on your version and I can do the rest.

**Reply by alienatedsec** [reply]
@0rangutan I just updated the repo to reference your work - thanks again

**Reply by ktakodra** [reply]
I was looking at this project. But now dataloggers are coming as usb so means converting to 4 pin break out then back to usb …has anyone done this ? I can’t seem to find the official adapters as discussed here https://youtu.be/sajg7smvXY0?si=cTkA6JOfSBS7Tr6G
